cranelift-codegen/src/binemit/memorysink.rs

```diff
@@ -34,6 +34,12 @@ pub struct MemoryCodeSink<'a> {
     offset: isize,
     /// Size of the machine code portion of output
     pub code_size: isize,
+    /// Size of jumptable portion of output (following code)
+    pub jumptables_size: isize,
+    /// Size of read-only data portion of output (following jump tables)
+    pub rodata_size: isize,
+    /// Size of everything
+    pub total_size: isize,
```

Comment:
Maybe add a `CodeOffsets` field instead.

Reply:
Yeah, I was pondering that but then I saw "isize" which is signed and I started worrying about semantics of overflow/underflow.  Memorysink is small but these are public fields, so I'll have to worry about all the consumers here.  I'll investigate.

Which reminds me, how much do we know about consumers of Cranelift outside of Firefox?  Is everyone listening in on the traffic in this repo?

Reply:
A size can never be negative right? So it should probably use an unsigned integer.

> how much do we know about consumers of Cranelift outside of Firefox? 

That I am one :)

Reply:
Well, no, a size can never be negative but the difference of two sizes can be.

Reply:
OK, updated this and cleaned up MemoryCodeSink a little in the process.  The offset field remains isize because its dominant use requires that form but I've commented to clarify this.  The structure is now called CodeInfo which is generic but more appropriate than CodeOffsets.

Reply:
Turns out MemoryCodeSink can be simplified further, more updates coming.